tests/test_rds.py

```diff
@@ -197,6 +197,83 @@ def test_rds_delete(self):
+            {'name': 'rds-upgrade-available',
+             'resource': 'rds',
+             'filters': [
+                 {'type': 'upgrade-available'},
+                 {'AutoMinorVersionUpgrade': False}
+             ],
+             'actions': [{
+                 'type': 'mark-for-op',
+                 'tag': 'custodian_upgrade',
+                 'days': 1,
+                 'msg': 'Minor engine upgrade available: {op}@{action_date}',
+                 'op': 'upgrade-minor'}],
+             }, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
+
+    def test_rds_minor_upgrade_match(self):
+        session_factory = self.replay_flight_data(
+            'test_rds_minor_upgrade_match')
+        p = self.load_policy(
+            {'name': 'rds-upgrade-list',
+             'resource': 'rds',
+             'filters': [
+                 {'type': 'marked-for-op', 'tag': 'custodian_upgrade',
+                  'op': 'upgrade-minor'}]}, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    def test_rds_minor_upgrade_do(self):
+        session_factory = self.replay_flight_data(
+            'test_rds_minor_upgrade_do')
+        p = self.load_policy(
+            {'name': 'rds-upgrade-do',
+             'resource': 'rds',
+             'filters': [
+                 'upgrade-available',
+                 {'type': 'marked-for-op', 'tag': 'custodian_upgrade',
+                  'op': 'upgrade-minor'}],
+             'actions': [{
+                 'type': 'upgrade-minor',
+                 'immediate': False}]}, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
+
+    def test_rds_minor_upgrade_unavailable(self):
+        session_factory = self.replay_flight_data(
+            'test_rds_minor_upgrade_unavailable')
+        p = self.load_policy(
+            {'name': 'rds-upgrade-done',
+             'resource': 'rds',
+             'filters': [
+                 {'type': 'upgrade-available', 'value': False}
+             ]}, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    def test_rds_minor_upgrade_immediate(self):
+        session_factory = self.replay_flight_data(
+            'test_rds_minor_upgrade_immediate')
+        p = self.load_policy(
+            {'name': 'rds-upgrade-do',
+             'resource': 'rds',
+             'filters': [
+                 {'AutoMinorVersionUpgrade': False},
+                 {'type': 'marked-for-op', 'tag': 'custodian_upgrade',
+                  'op': 'upgrade-minor'}],
+             'actions': [
+                 {'type': 'upgrade-minor', 'immediate': True},
+                 {'type': 'remove-tag', 'tags': ['custodian_upgrade']}
+             ]}, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
```

Comment:
you actually want to re-fetch the db to verify that the upgrade is in progress, ie. without testing that your effectively just testing that the filters ran and the action didn't error, your not testing the effect of the action.